defs.bzl

```diff
@@ -60,21 +51,30 @@ def buildifier_prebuilt_deps():
         sha256 = "c6966ec828da198c5d9adbaa94c05e3a1c7f21bd012a0b29ba8ddbccb2c93b0d",
     )
 
-    for platform, arch, buildifier_sha256, buildozer_sha256 in _BINARIES:
-        http_file(
-            name = "buildifier_{}_{}".format(platform, arch),
-            urls = ["https://github.com/bazelbuild/buildtools/releases/download/{}/buildifier-{}-{}".format(_VERSION, platform, arch)],
-            sha256 = buildifier_sha256,
-            downloaded_file_path = "buildifier",
-            executable = True,
-        )
-        http_file(
-            name = "buildozer_{}_{}".format(platform, arch),
-            urls = ["https://github.com/bazelbuild/buildtools/releases/download/{}/buildozer-{}-{}".format(_VERSION, platform, arch)],
-            sha256 = buildozer_sha256,
-            downloaded_file_path = "buildozer",
-            executable = True,
-        )
+    if len(assets) == 0:
+        fail("No assets were specified.")
+
+    for asset in assets:
+        http_file_args = {
+            "name": "{name}_{platform}_{arch}".format(
+                name = asset.name,
+                platform = asset.platform,
+                arch = asset.arch,
+            ),
+            "urls": [
+                "https://github.com/bazelbuild/buildtools/releases/download/{version}/{name}-{platform}-{arch}".format(
+                    version = asset.version,
+                    name = asset.name,
+                    platform = asset.platform,
+                    arch = asset.arch,
+                ),
+            ],
+            "downloaded_file_path": asset.name,
+            "executable": True,
+        }
+        if asset.sha256:
+            http_file_args["sha256"] = asset.sha256
```

Comment:
Huh I didn't know this could be optional

Reply:
I believe it can be omitted. I will confirm, once I finished the plumbing.

Reply:
The `specify_assets` example workspace and related integration tests confirm that the sha256 is optional. However, `http_file` will output a message stating how to make the declaration canonical.